### PR TITLE
Remove sensitive information from logs

### PR DESCRIPTION
**Pull request by cbartz**
Applicable spec: n/a

### Overview

1. Redact the token value in the previous state log. Also change the log level to debug for this particular log.

2. Hide the "ps aux" command as it can expose sensitive information from the workload.

### Rationale

1. The token is a secret and should not appear in the logs.
2. Workload can contain sensitive information which should not be logged.
### Juju Events Changes

n/a

### Module Changes

`charm_state.CharmState`: Added a helper function that redacts the token value before logging.
`runner_manager.RunnerManager._get_runner_health_states`: add `hide_cmd=True`

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Review by yanksyoon — DISMISSED** [review]

**Review by weiiwang01 — DISMISSED** [review]

**Review by yanksyoon — DISMISSED** [review]

File: src/charm_state.py [hunk 1]
```diff
@@ -1071,6 +1072,25 @@ def _check_immutable_config_change(
...
+        if logger.isEnabledFor(logging.DEBUG):
+            prev_state_for_logging = prev_state_dict.copy()
+            charm_config = prev_state_for_logging.get("charm_config")
+            if charm_config and "token" in charm_config:
+                charm_config = charm_config.copy()
+                charm_config["token"] = "*****"  # nosec
+            prev_state_for_logging["charm_config"] = charm_config
+
+            logger.debug("Previous charm state: %s", prev_state_for_logging)
+
```
Comment (yhaliaw) [inline src/charm_state.py:1093]: This might not hide all token during integration tests, as we supply multiple tokens during integration test and only pick one to use. There are also `alt-token` in integration tests.
For the `tox` process these tokens would still be visible.

Comment (cbartz) [reply src/charm_state.py:1093]: I think the charm state only contains the `token` config. You are referring to the `tox` command line which may be in the runner's workload, which should be hidden using the `hide_cmd=True` setting for the `ps aux` command.

**Review by yanksyoon — DISMISSED** [review]

**Review by weiiwang01 — DISMISSED** [review]

**Review by yanksyoon — DISMISSED** [review]

**Review by weiiwang01 — DISMISSED** [review]

**Review by yhaliaw — DISMISSED** [review]

**Review by yanksyoon — DISMISSED** [review]

**Review by weiiwang01 — DISMISSED** [review]

**Review by merkata — DISMISSED** [review]
LGTM
minor nit - the test assumes the method is a dunder method (two underscores) where it is a private method (one underscore)

**Review by yanksyoon — APPROVED** [review]

**Review by weiiwang01 — APPROVED** [review]

**Review by merkata — APPROVED** [review]
LGTM
